### PR TITLE
AA-2 Edit company init

### PR DESCRIPTION
**Pull request by theninjacoder-uz**

### CODE REVIEW

File: src/config_service/storage/postgres/config.go [hunk 1]
```diff
@@ -31,3 +32,25 @@ func (r *configRepo) CreateCompany(ctx context.Context, req entity.Company) (ent
 
 	return company, nil
 }
+
+func (r *configRepo) EditCompany(ctx context.Context, req entity.Company) error {
+	var exist bool
+
+	err := r.db.GetContext(ctx, exist, `SELECT true FROM companies WHERE id=$1`, req.Id)
```
Comment (khdoba2000) [inline src/config_service/storage/postgres/config.go:39]: why you are checking like this. You can get not found info from Update statement too.

**Review by khdoba2000 — CHANGES_REQUESTED** [review]

File: src/config_service/storage/postgres/config.go (hunk 1 above)
Comment (theninjacoder-uz) [reply src/config_service/storage/postgres/config.go:39]: Sunbula said that ExecContext(`Update ...`) not return error if row not found

File: src/config_service/service/company.go [hunk 2]
```diff
@@ -59,3 +61,37 @@ func (c *ConfigService) CreateCompany(ctx context.Context, req *pb.CreateCompany
...
+		c.logger.For(ctx).Error("failed to parse company id from string to uuid", zap.Error(err))
+		return &pb.Empty{}, status.Error(codes.Internal, err.Error())
+	}
+
+	err = c.storage.Configuration().EditCompany(ctx, *company)
+
+	if err != nil {
+		if errors.Is(err, constants.ErrCompanyNotExists) {
+			c.logger.For(ctx).Error("company not found with the given ID", zap.Error(err))
+			return &pb.Empty{}, status.Error(codes.NotFound, err.Error())
```
Comment (khdoba2000) [inline src/config_service/service/company.go:85]: for pb.Empty return nil instead

File: src/config_service/pkg/mappers/mapper.go [hunk 3]
```diff
@@ -23,3 +25,18 @@ func FromCompanyToProtoCompanyResponse(req entity.Company) *pb.CompanyResponse {
 		UpdateDate: req.UpdateDate.String(),
 	}
 }
+
+func MapProtoToUpdateCompanyRequest(req *pb.UpdateCompanyRequest) (*entity.Company, error) {
+	id, err := uuid.Parse(req.Id)
+
+	if err != nil {
+		return &entity.Company{}, nil
```
Comment (khdoba2000) [inline src/config_service/pkg/mappers/mapper.go:33]: You will lose the error of validating req.ID here. Should be "return nil, err"

File: src/config_service/service/company.go (hunk 2 above)
Comment (khdoba2000) [reply src/config_service/service/company.go:85]: for all pointer expected returns you should return nil if any error arises during operations.

File: src/config_service/storage/postgres/company.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+func (r *configRepo) CreateCompany(ctx context.Context, req entity.Company) (entity.Company, error) {
+	tx, err := r.db.BeginTxx(ctx, nil)
+	if err != nil {
+		tx.Rollback()
+		return entity.Company{}, err
+	}
+	var company entity.Company
+	if err := tx.QueryRowContext(
```
Comment (khdoba2000) [inline src/config_service/storage/postgres/company.go:28]: I said earlier to use Execute for insert update queries.

File: src/config_service/storage/postgres/company.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	tx, err := r.db.BeginTxx(ctx, nil)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	var count int64
+	// Update row and get the number of updated rows
+	if err := tx.QueryRowContext(ctx,
+		`WITH rows AS (
```
Comment (khdoba2000) [inline src/config_service/storage/postgres/company.go:57]: you still did not get my point, remove with rows as. just do "UPDATE ******** RETURNING *" this query returns rows and error(if any).  and then You can get length of rows in **go code** not is postgres.

File: src/config_service/storage/postgres/company.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+			name=$1  
+			logo=$2  
+			domain=$3  
+			update_date=$4
+			WHERE id=$5
+			RETURNING 1
+		)
+		SELECT count(*) FROM rows`,
+		req.Name, req.Logo, req.Domain, time.Now(), req.Id,
+	).Scan(count); err != nil {
```
Comment (khdoba2000) [inline src/config_service/storage/postgres/company.go:68]: you will not need to scan after above changes.

File: src/config_service/storage/postgres/company.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+		tx.Rollback()
+		return err
+	}
+	if count != 1 {
```
Comment (khdoba2000) [inline src/config_service/storage/postgres/company.go:72]: check will be like that "if len(rows)==0"

**Review by khdoba2000 — CHANGES_REQUESTED** [review]

File: src/config_service/service/company.go [hunk 8]
```diff
@@ -74,24 +75,24 @@ func (c ConfigService) EditCompany(ctx context.Context, req *pb.UpdateCompanyReq
...
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
 	c.logger.For(ctx).Info("Company create finished",
 		zap.String("Name", req.Name),
 		zap.String("Logo", req.Logo),
 		zap.String("Domain", req.Domain),
 	)
-	return &pb.Empty{}, nil
+	return nil, nil
```
Comment (khdoba2000) [inline src/config_service/service/company.go:97]: here the prev version was reasonable, there is concept that if function returns with some error it should return nil, 0, "" values for the expected result.

**Review by khdoba2000 — APPROVED** [review]

**Pull request merged**